prow/plugins/welcome/welcome.go

```diff
@@ -96,8 +96,8 @@ func handlePR(c client, pre github.PullRequestEvent, welcomeTemplate string) err
 		return err
 	}
 
-	// if there is exactly one result, this is the first! post the welcome comment
-	if len(issues) == 1 {
+	// if there are no results, this is the first! post the welcome comment
+	if len(issues) == 0 || len(issues) == 1 && issues[0].Number == pre.Number {
```

Comment:
btw, i think this is doable with `author_association == FIRST_TIME_CONTRIBUTOR`
https://github.com/google/go-github/blob/master/github/pulls.go#L59
https://developer.github.com/v4/enum/commentauthorassociation/

Reply:
that's amazing! I dug around a bunch when we first looked at this and didn't see this. I'll file an issue to switch, thanks!